.github/workflows/windows.yml

```diff
@@ -17,487 +17,451 @@ concurrency:
         run: |
           . "${{ env.OV_INSTALL_DIR }}/setupvars.ps1"
-          cmake -DCMAKE_BUILD_TYPE=Release -S ./ -B ./build/
-          cmake --build ./build/ --config Release --target py_openvino_genai --parallel --verbose
-        env:
-          CMAKE_TLS_VERIFY: 0
+          cmake -DCMAKE_BUILD_TYPE=${{ matrix.build-type }} -S ${{ env.OV_INSTALL_DIR }}/samples/cpp -B ${{ env.BUILD_DIR }}
+          cmake --build ${{ env.BUILD_DIR }} --config ${{ matrix.build-type }} --parallel $ENV:NUMBER_OF_PROCESSORS
+          cmake --install ${{ env.BUILD_DIR }} --config ${{ matrix.build-type }} --component samples_bin --prefix ${{ env.INSTALL_DIR }}
 
-      - name: Test bindings
-        run: |
-          . "${{ env.OV_INSTALL_DIR }}/setupvars.ps1"
-          python -m pip install ./thirdparty/openvino_tokenizers/[transformers] -r ./tests/python_tests/requirements.txt --find-links ${env:OV_INSTALL_DIR}/wheels
-          python -m pytest -v ./tests/python_tests/test_whisper_pipeline.py -k test_smoke
-        env:
-          PYTHONPATH: "./build/" # cmd evaluates variables in a different way. Setting PYTHONPATH before setupvars.bat instead of doing that after solves that.
+      - name: Upload Samples Build Package
+        if: always()
+        uses: actions/upload-artifact@b4b15b8c7c6ac21ea08fcf65892d2ee8f75cf882 # v4.4.3
+        with:
+          name: genai_samples_${{ matrix.build-type }}
+          path: ${{ env.INSTALL_DIR }}
+          if-no-files-found: 'error'
 
-      - name: Test bindings (wheel)
-        run: |
-          . "${{ env.OV_INSTALL_DIR }}/setupvars.ps1"
-          python -m pip install . --verbose --find-links ${env:OV_INSTALL_DIR}/wheels
-          python -m pip install ./tools/who_what_benchmark --find-links ${env:OV_INSTALL_DIR}/wheels
-          
-          # transformers 4.46.3 will enable return_timestamps tests
-          # this check enabled for windows only. Ticket: 160205.
-          python -m pip install transformers==4.46.3
-          
-          python -m pytest -v ./tests/python_tests/test_whisper_pipeline.py ./tests/python_tests/test_whisper_pipeline_static.py -k "not test_smoke"
-
-  genai_python_lib_vlm:
-    name: OpenVINO genai VLM tests (cmake + wheel)
-    needs: [ openvino_download, openvino_build ]
-    if: |
-      always() &&
-      (needs.openvino_download.outputs.status == 'success' || needs.openvino_build.result == 'success')
-    timeout-minutes: 90
+  genai_tests_wheel:
+    name: Python (${{ matrix.test.name}}) Tests (wheel)
+    needs: [ openvino_download, genai_build_wheel ]
+    timeout-minutes: 60
+    strategy:
+      fail-fast: false
+      matrix:
+        test:
+          - name: 'Whisper'
+            cmd: 'tests/python_tests/test_whisper_pipeline.py tests/python_tests/test_whisper_pipeline_static.py'
+          - name: 'LLM && VLM'
+            cmd: 'tests/python_tests --ignore tests/python_tests/test_whisper_pipeline.py --ignore tests/python_tests/test_whisper_pipeline_static.py'
```

Comment:
on Linux we have more groups now
https://github.com/openvinotoolkit/openvino.genai/blob/bfa9d3aee89444085432cd7195626aebf3fa4c9e/.github/workflows/linux.yml#L270-L275

Reply:
I don't mind to extend